### ISSUE TITLE
I had problem fine-tuned model inference.

### ISSUE THREAD

**Issue by Ahnsungmoon** [DepthAnything/Depth-Anything-V2#100]
Hello, Thanks for your work!!
I had problem fine-tuned model inferecing..
![Screenshot from 2024-07-22 19-28-58](https://github.com/user-attachments/assets/68eb4525-332c-4b9a-80da-f45682983efe)

How to resolve this problem?? I used the "depth_anything_v2_vitl.pth" pretrained model for own datasets.

**Comment by jorgejgnz** [comment]
I'm having the same problem for fine-tuned model of the metric depth version of DAv2. Parameters saved using `model.state_dict()` during fine-tuning don't match the parameter names expected by DepthAnythingV2 class.

Parameters `pretrained.x`, `depth_head.x` and `module.pretrained.x` are missing when using
`torch.load("exp/.../latest.pth", ...)['model']`

**Comment by jorgejgnz** [comment]
Solved! Generated state dict during fine-tuning adds _'module.'_ prefix to every param name and you have to remove that prefix for every param name before loading weights:

```python
model_state_dict = torch.load(args.load_from, map_location='cpu')['model']
model_state_dict = {k.replace('module.', ''): v for k, v in model_state_dict.items()}
depth_anything.load_state_dict(model_state_dict)
```

**Comment by Ahnsungmoon** [comment]
@jorgejgnz Works!! Thanks a lot !!
